### PR TITLE
Implement concrete classes before rename

### PR DESCRIPTION
**Pull request by UncleGrandpa925**

### CODE REVIEW

**Comment by UncleGrandpa925** [comment]
@simonteozw can you review this commit then merge it for me so that the next PR won't be so huge

**Review by simonteozw — COMMENTED** [review]
Ok looks good! I only have 1 comment: I think we should rename to `common` and not `commons`. Standardise to all subpackage names are singular, not plural.

Also, are we going to merge in the disabled checkstyle? I am afraid if that will cause code quality to drop / dont write sufficient test cases.

**Comment by UncleGrandpa925** [comment]
Actually "commons" in the name of one of the current module too! (seedu.address.commons)

**Comment by UncleGrandpa925** [comment]
Ah that checkstyle, I think it can be disabled for now, later on when I merge the later pull request I can enable it back again! Actually we can do cherry-pick there but I haven't had time to properly learn cherry-pick

**Review by simonteozw — COMMENTED** [review]
Ok Looks good! Just 1 question

File: src/main/java/seedu/address/model/module/Module.java [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+/**
+ * Represents a Lesson in the address book.
+ * Guarantees: details are present and not null, field values are validated, immutable.
+ */
+public class Module {
+    public static final String TYPE = "M";
+
+    private final Code code;
+    private final Name title;
+    private final String gradeDist;
```
Comment (simonteozw) [inline src/main/java/seedu/address/model/module/Module.java:22]: Can I ask what is gradeDist? Is it grade distribution?

**Review by simonteozw — APPROVED** [review]
I think we are good to go!

**Pull request merged**